File: app/src/main/java/com/simpleshift/scheduler/domain/model/LeaveStrategy.kt
package com.simpleshift.scheduler.domain.model

import androidx.compose.runtime.Immutable
import java.time.LocalDate

@Immutable
data class LeaveStrategy(
    val leaveDays: Int,
    val totalBreakDays: Int,
    val leaveDates: List<LocalDate>,
    val breakStart: LocalDate,
    val breakEnd: LocalDate,
    val holidayOverlap: Int,
    val weekendOverlap: Int,
    val overlappingHolidayNames: List<String>,
    val efficiency: Float,
    val score: Float
)

File: flutter/lib/domain/models/leave_strategy.dart
/// 请假策略 — 对应 Android 版 LeaveStrategy.kt
class LeaveStrategy {
  final int leaveDays;
  final int totalBreakDays;
  final List<DateTime> leaveDates;
  final DateTime breakStart;
  final DateTime breakEnd;
  final int holidayOverlap;
  final int weekendOverlap;
  final List<String> overlappingHolidayNames;
  final double efficiency;
  final double score;

  const LeaveStrategy({
    required this.leaveDays,
    required this.totalBreakDays,
    required this.leaveDates,
    required this.breakStart,
    required this.breakEnd,
    required this.holidayOverlap,
    required this.weekendOverlap,
    required this.overlappingHolidayNames,
    required this.efficiency,
    required this.score,
  });
}
